rotkehlchen/tests/unit/decoders/test_aave_v3.py

```diff
@@ -960,3 +960,98 @@ def test_aave_v3_events_with_approval(database, polygon_pos_inquirer, polygon_po
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_fees)),
+            location_label=scroll_accounts[0],
+            notes=f'Burned {gas_fees} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=58,
+            timestamp=timestamp,
+            location=Location.SCROLL,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=EvmToken('eip155:534352/erc20:0xf301805bE1Df81102C957f6d4Ce29d2B8c056B2a'),
+            balance=Balance(amount=FVal('115792089237316195423570985008687907853269984665640564039457.574007891353058083')),
+            location_label=scroll_accounts[0],
+            notes='Set aScrWETH spending approval of 0x76111D2841b41B15e6F07fBae4796a82438D9c90 by 0xFF75A4B698E3Ec95E608ac0f22A03B8368E05F5D to 115792089237316195423570985008687907853269984665640564039457.574007891353058083',  # noqa: E501
+            counterparty=None,
+            address=weth_gateway,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=59,
+            timestamp=timestamp,
+            location=Location.SCROLL,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=EvmToken('eip155:534352/erc20:0xf301805bE1Df81102C957f6d4Ce29d2B8c056B2a'),
+            balance=Balance(amount=FVal(gained_amount)),
+            location_label=scroll_accounts[0],
+            notes=f'Receive {gained_amount} aScrWETH from AAVE v3',
+            counterparty=CPT_AAVE_V3,
+            address=ZERO_ADDRESS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=62,
+            timestamp=timestamp,
+            location=Location.SCROLL,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=EvmToken('eip155:534352/erc20:0x5300000000000000000000000000000000000004'),
+            balance=Balance(),
+            location_label=scroll_accounts[0],
+            notes='Disable WETH as collateral on AAVE v3',
+            counterparty=CPT_AAVE_V3,
+            address=weth_gateway,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=63,
+            timestamp=timestamp,
+            location=Location.SCROLL,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.RETURN_WRAPPED,
+            asset=EvmToken('eip155:534352/erc20:0xf301805bE1Df81102C957f6d4Ce29d2B8c056B2a'),
+            balance=Balance(amount=FVal(withdrawn_amount)),
+            location_label=scroll_accounts[0],
+            notes=f'Return {withdrawn_amount} aScrWETH to AAVE v3',
+            counterparty=CPT_AAVE_V3,
+            address=weth_gateway,
+        ), EvmEvent(
```

Comment:
and finally an in event.

Reply:
As discussed in discord and also confirmed in telegram for the case of aWETH there is an extra transferFrom that triggers the mint + transfer event to the user. We have two blocks in the transaction. One minting and then withdrawing and the second depends on the extra amount minted in the first so there is an order on the groups for this event. The in/out order is respected there